exercises/grains/GrainsTest.swift

```diff
@@ -3,51 +3,90 @@ import XCTest
 
 
 
-// Return multiple types and compare by overloading ==
 
 class GrainsTest: XCTestCase {
     
     func testInvalidInput1() {
-        XCTAssertTrue((result: 0, error: "Input[65] invalid. Input should be between 1 and 64 (inclusive)") == Grains.square(65))
+        var throwsInputTooHighError = false
+        
+        defer {
+            XCTAssertTrue(throwsInputTooHighError)
+        }
+        
+        do {
+            let _ = try Grains.square(65)
+        } catch Grains.Error.InputTooHigh(let message) {
+            throwsInputTooHighError = true
+            XCTAssertTrue(message == "Input[65] invalid. Input should be between 1 and 64 (inclusive)")
+        } catch {
+            return
+        }
+        
     }
     
     func testInvalidInput2() {
-        XCTAssertTrue((result: 0, error: "Input[0] invalid. Input should be between 1 and 64 (inclusive)") == Grains.square(0))
+        var throwsInputTooLowError = false
+        
+        defer {
+            XCTAssertTrue(throwsInputTooLowError)
+        }
+        
+        do {
+            let _ = try Grains.square(0)
+        } catch Grains.Error.InputTooLow(let message) {
+            throwsInputTooLowError = true
+            XCTAssertTrue(message == "Input[0] invalid. Input should be between 1 and 64 (inclusive)")
+        } catch {
```

Comment:
This block would only be executed if an error (other than `.inputTooLow`) is thrown, though. It would miss the path of the try actually succeeding. That was the reason for the `defer`.